File: example/func.lua
-- package: example

local gopkg = _G["github.com/edwingeng/go2lua/example"]
do
    local g = _G
    local newEnv = setmetatable({}, {
        __index = function (t, k)
            local v = gopkg[k]
            if v == nil then return g[k] end
            return v
        end,
        __newindex = gopkg,
    })
    _ENV = newEnv
end

func1 = function()
    local f1 = function ()
        print("f1")
    end
    f1()

    local f2 = function ()
        print("f2")
    end
    f2()

    local __lambda = function ()
        print("f3")
    end
    __lambda()

    local __lambda_x2 = function (n1, n2, n3)
        print(n1, n2, n3)
    end
    __lambda_x2(1, 2, 3)
end

func2 = function()
    local __lambda = function (cb)
        if cb ~= nil then
            cb(100)
        end
    end
    __lambda(function (n)
        print(n)
    end)
end

func3 = function()
    local a = 0
    local b, c = "", ""

    if a > 0 then
        return 100, "b", "c"
    end
    return a, b, c
end

return function() end

File: example/func.go
package example

func func1() {
	var f1 = func() {
		println("f1")
	}
	f1()

	f2 := func() {
		println("f2")
	}
	f2()

	func() {
		println("f3")
	}()

	func(n1, n2, n3 int) {
		println(n1, n2, n3)
	}(1, 2, 3)
}

func func2() {
	func(cb func(n int)) {
		if cb != nil {
			cb(100)
		}
	}(func(n int) {
		println(n)
	})
}
